atomicapp/providers/kubernetes.py

```diff
@@ -204,3 +205,36 @@ def undeploy(self):
             if self.config_file:
                 cmd.append("--kubeconfig=%s" % self.config_file)
             self._call(cmd)
+
+    """
+    All extra methods that belong to 'requirements'
+    Each method will be passed the respective graph from Nulecule
+    and any other extra information that is required.
+
+    Actions are either: run, stop or uninstall
+    """
+
+    def persistent_storage(self, graph, action):
+        logger.debug("Persistent storage enabled! Running action: %s" % action)
+
+        # Get the path of the persistent storage yaml file includes in /external
+        # Plug the information from the graph into the persistent storage file
+        base_path = os.path.dirname(os.path.realpath(__file__))
+        template_path = os.path.join(base_path, 'external/kubernetes/persistent_storage.yaml')
+        with open(template_path, 'r') as f:
+            content = f.read()
+            template = Template(content)
+            rendered_template = template.safe_substitute(graph)
+
+        # Must write persistent storage to a tmp file as to not have malicious intent
+        # when running direct cmd
+        tmp_file = Utils.getTmpFile(rendered_template, '.yaml')
+
+        # Pass the .yaml file and execute
+        if action is "run":
+            cmd = [self.kubectl, "create", "-f", tmp_file, "--namespace=%s" % self.namespace]
+            if self.config_file:
+                cmd.append("--kubeconfig=%s" % self.config_file)
+            self._call(cmd)
```

Comment:
Given this is "persistent" we probably don't want to ever be deleting. I believe OpenShift does not delete storage when all objects are destroyed. It's treated *special* so as to do no unexpected harm.